lib/task_helpers.rb

```diff
@@ -1,213 +1,106 @@
-        # we catch up once before the alias move and once afterwards.
-        first_catchup_start_time = Time.now
-        adjusted_start_time = initial_start_time - (extra_catchup_minutes * 60)
-        catchup_index(adjusted_start_time, index_name, batch_size, sleep_time)
+      first_catchup_start_time = Time.now
+      adjusted_start_time = first_catchup_start_time - extra_catchup_minutes * 60
+      catchup_indices(adjusted_start_time, batch_size)
 
-        move_alias(alias_name, index_name, force_delete: true)
-        adjusted_start_time = first_catchup_start_time - (extra_catchup_minutes * 60)
-        catchup_index(adjusted_start_time, alias_name, batch_size, sleep_time)
-      end
-
-      LOG.info "Rebuild index complete."
-      index_name
+      LOG.info "Rebuild indices complete."
     end
 
-    def self.catchup_index(start_time, index_name, batch_size=100, sleep_time=0)
-      [Comment, CommentThread].each do |model|
+    def self.catchup_indices(start_time, batch_size=100)
+      INDEX_MODELS.each do |model|
         current_batch = 1
-        model.where(:updated_at.gte => start_time).import(index: index_name, batch_size: batch_size) do |response|
-            batch_import_post_process(response, current_batch, sleep_time)
+        model.where(:updated_at.gte => start_time).import(index: model.index_name, batch_size: batch_size) do |response|
+            batch_import_post_process(response, current_batch)
             current_batch += 1
         end
       end
       LOG.info "Catch up from #{start_time} complete."
     end
 
-    def self.create_index(name=nil)
-      name ||= "#{Content::ES_INDEX_NAME}_#{Time.now.strftime('%Y%m%d%H%M%S%L')}"
-
-      Elasticsearch::Model.client.indices.create(index: name)
-      put_mappings(name)
-
-      LOG.info "Created new index: #{name}."
-      name
+    def self.create_indices
+      INDEX_MODELS.each do |model|
+        model.__elasticsearch__.create_index! force: true
+      end
+      LOG.info "Created new indices."
     end
 
-    def self.delete_index(name)
-      begin
-        Elasticsearch::Model.client.indices.delete(index: name)
-        LOG.info "Deleted index: #{name}."
-      rescue Elasticsearch::Transport::Transport::Errors::NotFound
-        # NOTE (CCB): Future versions of the Elasticsearch client support the ignore parameter,
-        # that can be used to ignore 404 errors.
-        LOG.info "Unable to delete non-existent index: #{name}."
-      end
+    def self.delete_indices
+      Elasticsearch::Model.client.indices.delete(index: INDEX_NAMES, ignore_unavailable: true)
+      LOG.info "Delete indices."
     end
 
-    def self.batch_import_post_process(response, batch_number, sleep_time)
-        response['items'].select { |i| i['index']['error'] }.each do |item|
-            LOG.error "Error indexing. Response was: #{response}"
-        end
-        LOG.info "Imported batch #{batch_number} into the index"
-        sleep(sleep_time)
```

Comment:
Why is it we no longer need to sleep here?  (I'm pre-supposing there was a good reason, but maybe there wasn't?)

Reply:
When we load data to the index ES is acknowledged that index was updated. ES periodically refresh index.
[Indices refresh description doc.](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html#refresh-api-desc)